### PR TITLE
Add missing onCommitSuccess/onCommitFailure to XATransaction 

### PR DESCRIPTION
**Pull request by ahmetmircik**
addressed this comment: https://github.com/hazelcast/hazelcast/pull/16581/files#r391776655

### CODE REVIEW

**Comment by mmedenjak** [comment]
Can you add a test for this and other PRs covering this scenario?

**Comment by ahmetmircik** [comment]
@mmedenjak done

File: hazelcast/src/test/java/com/hazelcast/xa/NearCacheXATest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            TransactionalMap txnMap = ctx.getMap(mapName);
+            valueReadInsideTxnFromTxnMapBeforeUpdate = txnMap.get(key);
+
+            txnMap.put(key, updatedValue);
+
+            valueReadInsideTxnFromTxnMapAfterUpdate = txnMap.get(key);
+            valueReadInsideTxnFromNonTxnMapAfterUpdate = map.get(key);
+        } catch (Exception e) {
+            e.printStackTrace();
+            error = true;
```
Comment (petrpleshachkov) [inline hazelcast/src/test/java/com/hazelcast/xa/NearCacheXATest.java]: Shall we fail the test here?

Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/xa/NearCacheXATest.java]: done

**Pull request merged**